realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -556,6 +571,8 @@ private void emitCodeForUnderConstruction(JavaWriter writer, boolean isPrimaryKe
                 .emitEmptyLine();
     }
 
+    // Note that because of bytecode hackery, this method may run before the constructor!
+    // It may ever run before fields have been initialized.
```

Comment:
Also, I don't believe it can run _before_ the constructor,  but it will be invoked during the constructor. But I guess the effect is the same. Fields might not initialized.